test/unit/middlewares/subscription-validator.test.js

```diff
@@ -0,0 +1,91 @@
+const Sinon = require("sinon");
+const { expect } = require("chai");
+const { validateSubscribe } = require("../../../middlewares/validators/subscription");
+
+describe("Middleware | Validators | Subscription", function () {
+  let req, res, nextSpy;
+
+  beforeEach(function () {
+    req = { body: {} };
+    res = {
+      status: Sinon.stub().returnsThis(),
+      json: Sinon.stub(),
+    };
+    nextSpy = Sinon.spy();
+  });
+
+  it("should call next function when a valid request body is passed", async function () {
+    req.body = {
+      phoneNumber: "+911234567890",
+      email: "test@example.com",
+    };
+
+    await validateSubscribe(req, res, nextSpy);
+
+    expect(nextSpy.calledOnce).to.be.equal(true);
+    expect(res.status.called).to.be.equal(false);
+    expect(res.json.called).to.be.equal(false);
+  });
+
+  it("should not return an error when phoneNumber is missing", async function () {
+    req.body = {
+      email: "test@example.com",
+    };
+
+    await validateSubscribe(req, res, nextSpy);
+    expect(nextSpy.calledOnce).to.be.equal(true);
+    expect(res.status.called).to.be.equal(false);
+    expect(res.json.called).to.be.equal(false);
+  });
+
+  it("should return a 400 error when email is missing", async function () {
+    req.body = {
+      phoneNumber: "+911234567890",
+    };
+
+    await validateSubscribe(req, res, nextSpy);
+
+    expect(nextSpy.called).to.be.equal(false);
+    expect(res.status.calledOnceWith(400)).to.be.equal(true);
+    expect(res.json.calledOnce).to.be.equal(true);
+    expect(res.json.firstCall.args[0]).to.have.property("error").that.includes('"email" is required');
+  });
+
+  it("should return a 400 error when both phoneNumber and email are missing", async function () {
+    req.body = {};
+
+    await validateSubscribe(req, res, nextSpy);
+    expect(nextSpy.called).to.be.equal(false);
+    expect(res.status.calledOnceWith(400)).to.be.equal(true);
+    expect(res.json.calledOnce).to.be.equal(true);
+    expect(res.json.firstCall.args[0]).to.have.property("error").that.includes('"email" is required');
+  });
+
+  it("should return a 400 error when email is not in correct format", async function () {
+    req.body = {
+      phoneNumber: "+911234567890",
+      email: "invalid-email",
+    };
+
+    await validateSubscribe(req, res, nextSpy);
+
+    expect(nextSpy.called).to.be.equal(false);
+    expect(res.status.calledOnceWith(400)).to.be.equal(true);
+    expect(res.json.calledOnce).to.be.equal(true);
+    expect(res.json.firstCall.args[0])
+      .to.have.property("error")
+      .that.includes('"email" with value "invalid-email" fails to match the required pattern');
+  });
+
+  it("should not return an error when phoneNumber is in correct format", async function () {
```

Comment:
what about should not return an error when phoneNumber is not into  correct format

Reply:
Sir, I have added test for sanitize as well as incorrect format phone number as well.